core/dbt/adapters/base/impl.py

```diff
@@ -1298,6 +1298,44 @@ def render_column_constraint(cls, constraint: ColumnLevelConstraint) -> str:
         else:
             return ""
 
+    @classmethod
+    def _parse_model_constraint(cls, raw_constraint: Dict[str, Any]) -> ModelLevelConstraint:
+        try:
+            ModelLevelConstraint.validate(raw_constraint)
+            c = ModelLevelConstraint.from_dict(raw_constraint)
+            return c
+        except Exception:
+            raise DbtValidationError(f"Could not parse constraint: {raw_constraint}")
+
+    @available
+    @classmethod
+    def render_raw_model_constraints(cls, raw_constraints: List[Dict[str, Any]]) -> List[str]:
+        return [c for c in map(cls.render_raw_model_constraint, raw_constraints) if c is not None]
+
+    @classmethod
+    def render_raw_model_constraint(cls, raw_constraint: Dict[str, Any]) -> Optional[str]:
+        constraint = cls._parse_model_constraint(raw_constraint)
+        return cls.render_model_constraint(constraint)
+
+    @classmethod
+    def render_model_constraint(cls, constraint: ModelLevelConstraint) -> Optional[str]:
+        """Render the given constraint as DDL text. Should be overriden by adapters which need custom constraint
+        rendering."""
+        constraint_prefix = f"constraint {constraint.name} " if constraint.name else ""
+        column_list = ", ".join(constraint.columns)
```

Comment:
Haven't looked over all of the adapter PR's, I'm assuming we can trust that these predicates are fairly consistent across warehouses. Are we expecting to add more in the near future or are these a canonical subset that we've chosen to support?

Reply:
We're not expecting to add more `ConstraintType`s in the near future - these are the set we've chosen to support based on current warehouse support. More discussion on `ConstraintType` and the `Constraint` here: https://github.com/dbt-labs/dbt-core/issues/6750

This method (`render_model_constraint`), as well as `render_column_constraint` would be overridable in specific adapter implementations, but this base is pretty consistent across implementations thus far.